integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/ExtensionIT.java

```diff
@@ -99,19 +99,20 @@ public void predicate_fromJsonString() {
 		SearchQuery<DocumentReference> query = searchTarget.query( sessionContext )
 				.asReferences()
 				.predicate().bool( b -> {
-					b.should().withExtensionOptional(
+					b.should().extension().ifSupported(
 							LuceneExtension.get(),
 							// FIXME find some way to forbid using the context twice... ?
 							c -> c.fromLuceneQuery( new TermQuery( new Term( "string", "text 1" ) ) )
 					);
```

Comment:
Is it valid to call `ifSupported()` without having a `orElse()` or `orElseFail()`?

Reply:
Right... so, two problems:

1. I forgot to change this test to use the simpler `extension( MyExtension.get() ).fromLuceneQuery( ... )` syntax like I did in the Elasticsearch test. Fixed.
2. I forgot to test the very case you're mentioning, which should ideally trigger an error at some point. I'll see what I can do on Monday.

Reply:
> which should ideally trigger an error at some point

I mean, it probably should if the extension was not supported.

Reply:
It was (much) easier to allow the syntax `ifSupported()` without a following `orElse`, so I added some tests and allowed it.